code/modules/hydroponics/trays/tray_reagents.dm

```diff
@@ -85,35 +85,87 @@
 // *************************************
 
 /obj/item/reagent_containers/glass/fertilizer
-	name = "fertilizer bottle"
-	desc = "A small glass bottle. Can hold up to 60 units."
+	name = "jug"
+	desc = "A decent sized plastic jug. Can hold up to 80 units."
 	icon = 'icons/obj/chemical.dmi'
-	icon_state = "bottle16"
-	flags = OPENCONTAINER
-	possible_transfer_amounts = null
-	w_class = ITEMSIZE_SMALL
+	icon_state = "plastic_jug"
+	item_state = "plastic_jug"
+	flags = 0
+	possible_transfer_amounts = list(5, 10, 20, 40, 80)
+	volume = 80
+	w_class = ITEMSIZE_NORMAL
+
+	hitsound = 'sound/weapons/jug_empty_impact.ogg'
+	drop_sound = 'sound/weapons/jug_empty_impact.ogg'
 
-	//Like a shot glass!
-	amount_per_transfer_from_this = 10
-	volume = 60
+	fragile = 0
+	shatter = FALSE
+	unacidable = FALSE
+
+	force = 1
+	throwforce = 1
 
 /obj/item/reagent_containers/glass/fertilizer/Initialize()
 	. = ..()
 
 	src.pixel_x = rand(-5.0, 5)
 	src.pixel_y = rand(-5.0, 5)
 
+/obj/item/reagent_containers/glass/fertilizer/on_reagent_change()
+	. = ..()
+	update_icon()
+	if(reagents.total_volume)
+		var/fraction = reagents.total_volume / volume
+		force = max(10 * fraction, 1)
+		throwforce = max(10 * fraction, 1)
+		hitsound = 'sound/weapons/jug_filled_impact.ogg'
+		drop_sound = 'sound/weapons/jug_filled_impact.ogg'
+	else
+		force = 1
+		throwforce = 1
+		hitsound = 'sound/weapons/jug_empty_impact.ogg'
+		drop_sound = 'sound/weapons/jug_empty_impact.ogg'
+
+/obj/item/reagent_containers/glass/fertilizer/update_icon()
+	cut_overlays()
+
+	if(reagents.total_volume)
+		var/image/filling = image('icons/obj/reagentfillings.dmi', src, "plastic_jug10")
+
+		var/percent = round((reagents.total_volume / volume) * 100)
+		switch(percent)
+			if(0 to 10)
+				filling.icon_state = "plastic_jug-10"
+			if(11 to 29)
+				filling.icon_state = "plastic_jug25"
+			if(30 to 45)
+				filling.icon_state = "plastic_jug40"
+			if(46 to 61)
+				filling.icon_state = "plastic_jug55"
+			if(62 to 77)
+				filling.icon_state = "plastic_jug70"
+			if(78 to 92)
+				filling.icon_state = "plastic_jug85"
+			if(93 to INFINITY)
+				filling.icon_state = "plastic_jug100"
```

Comment:
Not sure how it's done in other places but I _think_ it would be beneficial to show empty containers only when they are _completely_ empty and similarly with _completely_ full.

Reply:
Agreed. Do that.